### PR TITLE
Add collision handling

### PR DESCRIPTION
**Pull request by Daniel-ND**

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/GameView.java [hunk 1]
```diff
@@ -55,15 +64,23 @@ public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
...
 
+        // Handpicked value
+        // Droid should be on a ground height, but platform includes grass.
+        int groundHeight = platform.getHeight() / 5 * 3;
+
         // Create droid
-        droid = new Droid(screenMargin, screenY - screenMargin, getResources());
+        droid = new Droid(screenMargin, screenY - groundHeight, getResources());
+
+        // Create obstacle list
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameView.java:74]: nit: no need for the comment. The code is pretty explanatory.

File: app/src/main/java/com/google/droidjump/GameView.java [hunk 2]
```diff
@@ -132,6 +161,12 @@ public void drawScene() {
             // Drawing droid
             drawDroid(canvas);
 
+            // Drawing Obstacles
+            drawObstacles(canvas);
+
+            // Drawing platform
+            drawPlatform(canvas);
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameView.java:169]: nit:remove comments and new lines

File: app/src/main/java/com/google/droidjump/GameView.java [hunk 3]
```diff
@@ -154,17 +189,31 @@ public void pause() {
...
         findNavController(this).navigate(R.id.action_game_screen_to_game_success_screen);
     }
 
     private void drawDroid(Canvas canvas) {
         canvas.drawBitmap(droid.getBitmap(), droid.getX(), droid.getY(), /* paint= */ null);
     }
 
+    private void drawPlatform(Canvas canvas) {
+        int platformY = screenY - platform.getHeight();
+        for (int curPlatformX = platformX; curPlatformX < screenX; curPlatformX += platform.getWidth())
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameView.java:207]: nit: add { }

File: app/src/main/java/com/google/droidjump/GameView.java [hunk 4]
```diff
@@ -173,4 +222,18 @@ public boolean onTouchEvent(MotionEvent event) {
         }
         return super.onTouchEvent(event);
     }
+
+    public void handleCollision() {
+        for (Obstacle obstacle : obstacleList) {
+            if (checkIntersection(droid, obstacle))
+                failGame();
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameView.java:229]: add { }

File: app/src/main/java/com/google/droidjump/models/Bat.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.models;
+
+import android.content.res.Resources;
+import android.graphics.BitmapFactory;
+import com.google.droidjump.R;
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/Bat.java:22]: nit: add java doc for public class.

File: app/src/main/java/com/google/droidjump/models/Droid.java [hunk 6]
```diff
@@ -14,13 +14,15 @@
...
  */
 
-package com.google.droidjump;
+package com.google.droidjump.models;
 
 import android.content.res.Resources;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
+import com.google.droidjump.GameConstants;
+import com.google.droidjump.R;
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/Droid.java:24]: nit: add class documentation

File: app/src/main/java/com/google/droidjump/models/Obstacle.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.models;
+
+import android.content.res.Resources;
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/Obstacle.java:20]: same as above. Add java doc

File: app/src/main/java/com/google/droidjump/models/Palm.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.models;
+
+import android.content.res.Resources;
+import com.google.droidjump.R;
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/models/Palm.java:21]: same as above

File: app/src/main/java/com/google/droidjump/GameView.java (hunk 3 above)
Comment (Daniel-ND) [reply app/src/main/java/com/google/droidjump/GameView.java]: Done

File: app/src/main/java/com/google/droidjump/GameView.java (hunk 4 above)
Comment (Daniel-ND) [reply app/src/main/java/com/google/droidjump/GameView.java:279]: Done

**Pull request merged**